core/libp2p/host.cpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/host.hpp"
+
+namespace libp2p {
+
+  peer::PeerId Host::getId() const noexcept {
+    return id_;
+  }
+
+  gsl::span<const multi::Multiaddress> Host::getListenAddresses() {
+    return network_->getListenAddresses();
+  }
+
+  outcome::result<void> Host::connect(const peer::PeerInfo &p) {
+    // TODO: review this. this logic may not be here
```

Comment:
This TODO is not quite clear to me

Reply:
The point is that BV is not sure if we should throw away a returned shared_ptr to the connection if we want just to dial with the peer, which is the purpose of this method. I think that's a correct logic, but the actual answer will be revealed during the implementation